bob-common/src/data.rs

```diff
@@ -8,7 +8,63 @@ use std::{
     hash::Hash,
 };
 
-pub type BobKey = u64;
+include!(concat!(env!("OUT_DIR"), "/key_constants.rs"));
+
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug)]
+pub struct BobKey([u8; BOB_KEY_SIZE]);
+
+impl From<Vec<u8>> for BobKey {
+    fn from(v: Vec<u8>) -> Self {
+        if v.len() > BOB_KEY_SIZE {
+            panic!("expected key size {}, received {}", BOB_KEY_SIZE, v.len());
+        }
+        let mut data = [0; BOB_KEY_SIZE];
+        for (ind, elem) in v.into_iter().enumerate() {
```

Comment:
More efficient way to do this is via [`TryFrom`](https://doc.rust-lang.org/std/primitive.array.html#impl-TryFrom%3CVec%3CT%2C%20A%3E%3E) trait:
```
let data = v.try_into().expect("array/vec size mismatch");
```
To clone all buffer at once